sample_consensus/include/pcl/sample_consensus/sac_model.h

```diff
@@ -386,6 +386,16 @@ namespace pcl
         max_radius = radius_max_;
       }
 
+      /** \brief This can be used to impose any kind of constraint on the model,
+        * e.g. that it has a specific direction, size, or anything else.
+        * \param[in] function A function that gets model coefficients and returns whether the model is acceptable or not.
+        */
+      inline void
+      setModelConstraints (std::function<bool(const Eigen::VectorXf &)> function)
+      {
+        is_model_valid_user_defined_ = std::move (function);
```

Comment:
1. `custom_model_constraint_` or similar name to match with the function name
2. Add a check here for verifying that `function` is valid (aka not empty)